source/code/plugins/oms_common.rb

```diff
@@ -575,34 +712,22 @@ def get_os_version(conf_path = "/etc/opt/microsoft/scx/conf/scx-release")
         return @@OSVersion
       end
 
-      def get_hostname
-        return @@Hostname if !@@Hostname.nil?
+      def get_hostname(ignoreOldValue=false)
+        # Notes:  This implementation does not try to address multi-byte character aspects,
+        # but merely tries to address simple exigencies.  There may be multi-byte issues
+        # we will need to address in the future here.
+        if ! is_hostname_compliant?(@@Hostname) or ignoreOldValue then
 
-        # Issue:
-        #   When omsagent runs inside a container, gethostname returns the hostname of the container (random name)
-        #   not the actual machine hostname.
-        #   One way to solve this problem is to set the container hostname same as machine name, but this is not
-        #   possible when host-machine is a private VM inside a cluster.
-        # Solution:
-        #   Share/mount ‘/etc/hostname’ as '/var/opt/microsoft/omsagent/state/containername' with container and
-        #   omsagent will read hostname from shared file.
+            look_in_Designated_HostnameFile    if has_Designated_HostnameFile?
 
-        begin
-          if File.exist?(@@HostnameFilePath) && File.readable?(@@HostnameFilePath)
-            @@Hostname = File.read(@@HostnameFilePath).strip
-            return @@Hostname
-          end
-        rescue => error
-          OMS::Log.warn_once("Unable to read the hostname from #{@@HostnameFilePath}: #{error}")
-        end
+            look_for_sockety_host_address      unless is_hostname_compliant?(@@Hostname)
 
-        begin
-          hostname = Socket.gethostname.split(".")[0]
-        rescue => error
-          OMS::Log.error_once("Unable to get the Host Name: #{error}")
-        else
-          @@Hostname = hostname
+            unless is_hostname_compliant?(@@Hostname)
+            # Give up, and take whatever the shell's got.
+              @@Hostname = clean_hostname_string(`hostname`)
+            end
         end
+        validate_hostname_equivalent(@@Hostname)
```

Comment:
If this raises a non-OMS-specific error, FluentD will catch it and the data being handled to be sent to OMS will not be sent. If you intend to allow errors being thrown from this validate routine, then please test what happens to data that is being handled when the error is thrown. It is not acceptable if the data gets dropped.